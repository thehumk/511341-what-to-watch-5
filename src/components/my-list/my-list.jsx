import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {ActionCreator} from '../../store/action';
import FilmsList from '../films-list/films-list';
import Header from '../header/header';
import Footer from '../footer/footer';
import {propsForFilms} from '../../utils/prop-types';

const MyList = (props) => {
  const {films, renderFilmsCount, changeRenderFilmsCount} = props;

  return (
    <div className="user-page">
      <Header classHeader={`user-page__head`}>
        <h1 className="page-title user-page__title">My list</h1>
      </Header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>

        <FilmsList
          films={films}
          renderFilmsCount={renderFilmsCount}
          changeRenderFilmsCount={changeRenderFilmsCount}/>
      </section>

      <Footer/>
    </div>
  );
};

MyList.propTypes = {
  films: PropTypes.arrayOf(propsForFilms).isRequired,
  renderFilmsCount: PropTypes.number.isRequired,
  changeRenderFilmsCount: PropTypes.func.isRequired,
};

const mapStateToProps = ({DATA, FILMS_STATUS}) => ({
  films: DATA.films.filter((elem) => elem.is_favorite === true),
  renderFilmsCount: FILMS_STATUS.renderFilmsCount,
});

const mapDispatchToProps = (dispatch) => ({
  changeRenderFilmsCount(count) {
    dispatch(ActionCreator.changeRenderFilmsCount(count));
  }
});

export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
