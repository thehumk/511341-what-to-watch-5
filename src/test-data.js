export const noop = () => {};

export const MATCH = {params: {id: `1`}};

export const mockComments = [
  {
    comment: `123`,
    date: `2020-10-29T13:38:44.769Z`,
    id: 1,
    rating: 4.6,
    user: {
      id: 12,
      name: `Isaac`
    },
  },
  {
    comment: `1234`,
    date: `2020-10-29T13:38:45.769Z`,
    id: 2,
    rating: 4.5,
    user: {
      id: 13,
      name: `Isa`
    },
  }
];

/* eslint-disable */

export const mockFilms = [
  {
    background_color: `#73B39A`,
    background_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    id: 1,
    is_favorite: true,
    name: `Bronson`,
    poster_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 3.6,
    released: 2008,
    run_time: 92,
    scores_count: 109661,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    video_link: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  },
  {
    background_color: `#73B39A`,
    background_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    genre: `Drama`,
    id: 2,
    is_favorite: false,
    name: `Bronson`,
    poster_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 3.6,
    released: 2008,
    run_time: 92,
    scores_count: 109661,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    video_link: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  },
];

/* eslint-enable */

export const state = {
  DATA: {
    films: mockFilms,
    promoFilm: mockFilms[0],
    genresList: [`All`, `Action`, `Drama`],
    isApplicationReady: false,
  },
  FILMS_STATUS: {
    activeGenre: `All`,
    renderFilmsCount: 8,
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    user: null,
  },
  COMMENTS: {
    comments: mockComments,
  },
};
