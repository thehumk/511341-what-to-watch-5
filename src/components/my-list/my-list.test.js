import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import {MyList} from './my-list';
import {noop, mockFilms, state} from '../../test-data';

const mockStore = configureMockStore()(state);

it(`Should MyList render correctly`, () => {
  const tree = renderer.create(
      <Provider store={mockStore}>
        <BrowserRouter>
          <MyList
            films={mockFilms}
            renderFilmsCount={8}
            changeRenderFilmsCount={noop}
          />
        </BrowserRouter>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
