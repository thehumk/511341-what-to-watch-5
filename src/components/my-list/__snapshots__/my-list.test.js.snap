// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MyList render correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/films/1"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <span>
              Bronson
            </span>
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/films/2"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <span>
              Bronson
            </span>
          </h3>
        </a>
      </article>
    </div>
    <div
      className="catalog__more"
    >
      
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
